pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -2450,17 +2450,20 @@ def _enumerate_with_start(
         return False, confidence
 
     def _get_start_value(self, node: nodes.NodeNG) -> tuple[int | None, Confidence]:
-        if isinstance(node, (nodes.Name, nodes.Call, nodes.Attribute)) or (
-            isinstance(node, nodes.UnaryOp)
-            and isinstance(node.operand, (nodes.Attribute, nodes.Name))
-        ):
```

Comment:
We also avoid this call when we have a const or minus a const so my intuition is that it's going to be faster in the common use case, even if we infer more on the strange use case, but I'm not in the mood for a benchmark to prove it.